fastasyncworldedit/API/fill-region-by-pattern.md

```diff
@@ -0,0 +1,78 @@
+# Filling a region using a pattern
+
+## Foreword
+First of all, patterns usually come from the core part of Worleedit or FAWE. The patterns can be found in the following path: [worldedit-core/src/main/java/com/fastasyncworldedit/core/function/pattern](https://intellectualsites.github.io/fastasyncworldedit-javadocs/worldedit-core/com/fastasyncworldedit/core/extension/factory/parser/pattern/package-summary.html)
+
+_The link refers you to the JavaDocs with the correct package_
+
+In this example, we only refer to one pattern for the time being.
+
+## Biome Cuboid Example
```

Comment:
Biome Patterns are probably a bad example, as they are fundamentally different from other patterns - it would require an explanation why biomes are handled as patterns in FAWE. I think a simple example would be e.g. `Linear3DBlockPattern`. It is also worth mentioning that `BlockType` and `BlockState` already implement `Pattern`, which can be done when passing block types to the L3D constructor.

Reply:
I would currently keep the example, as I could take the concrete from the support. But I would add others in another PR

Reply:
It's a bad and rather confusing example. The mix of patterns (which are *normally* about blocks) with biomes isn't helpful for most people. If at all, biomes should be mentioned at the very end, with a note that this differs from all other patterns.

Reply:
Then I would do 1-2 more in the PR with other things. Could you give me some concrete ones that make sense?

Reply:
As I mentioned, `Linear3DBlockPattern` is a very good choice. It allows showing how a rather "complex" pattern can be mapped from the command to the code, using multiple parameters and nested patterns. There basically doesn't need to be more than that, because that's the core of how patterns work. The biome pattern is rather a hacky special case.